### PR TITLE
Add java.util.logging javaagent instrumentation

### PR DESCRIPTION
**Pull request by trask**
There's no autoconfiguration of log exporter (yet), so this instrumentation doesn't benefit end users unless they explicitly configure a logging exporter via javaagent extension (this is how the tests work).

### CODE REVIEW

File: instrumentation/java-util-logging/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jul/JavaUtilLoggingInstrumentationModule.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import java.util.List;
+
+@AutoService(InstrumentationModule.class)
+public class JavaUtilLoggingInstrumentationModule extends InstrumentationModule {
+  public JavaUtilLoggingInstrumentationModule() {
+    // this name is important currently because it's used to disable this instrumentation
+    super("java-util-logging-spans");
```
Comment (trask) [inline instrumentation/java-util-logging/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jul/JavaUtilLoggingInstrumentationModule.java:19]:
```suggestion
    super("java-util-logging");
```

**Review by anuraaga — APPROVED** [review]

File: instrumentation/java-util-logging/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jul/JavaUtilLoggingHelper.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import io.opentelemetry.instrumentation.api.appender.Severity;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Formatter;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+public class JavaUtilLoggingHelper {
```
Comment (anuraaga) [inline instrumentation/java-util-logging/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jul/JavaUtilLoggingHelper.java:23]:
```suggestion
public final class JavaUtilLoggingHelper {
```

File: instrumentation/java-util-logging/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jul/JavaUtilLoggingInstrumentation.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  @Override
+  public void transform(TypeTransformer transformer) {
+    transformer.applyAdviceToMethod(
+        isMethod()
+            .and(isPublic())
+            .and(named("log"))
+            .and(takesArguments(1))
+            .and(takesArgument(0, named("java.util.logging.LogRecord"))),
+        JavaUtilLoggingInstrumentation.class.getName() + "$LogAdvice");
+    transformer.applyAdviceToMethod(
```
[the remark this answers is not in the document]
Comment (trask) [reply instrumentation/java-util-logging/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jul/JavaUtilLoggingInstrumentation.java]: I removed the jboss instrumentation, I'm not sure if that should be (conveniently) wedged into this instrumentation, or if it should live in its own instrumentation module. Plus sending as a separate PR will allow better scrutiny of it (and I realized the jboss test I included doesn't actually explain or verify the jboss instrumentation 😭).

File: instrumentation/java-util-logging/javaagent/src/test/groovy/JavaUtilLoggingTest.groovy [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        assertThat(log.getAttributes().get(SemanticAttributes.EXCEPTION_MESSAGE)).isNull()
+        assertThat(log.getAttributes().get(SemanticAttributes.EXCEPTION_STACKTRACE)).isNull()
+      }
+      if (parent) {
+        assertThat(log.getSpanContext()).isEqualTo(traces.get(0).get(0).getSpanContext())
+      } else {
+        assertThat(log.getSpanContext().isValid()).isFalse()
+      }
+    } else {
+      Thread.sleep(500) // sleep a bit just to make sure no span is captured
```
Comment (mateuszrzeszutek) [inline instrumentation/java-util-logging/javaagent/src/test/groovy/JavaUtilLoggingTest.groovy:70]:
```suggestion
      Thread.sleep(500) // sleep a bit just to make sure no log is captured
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**